SetupBLT.cmake

```diff
@@ -45,6 +45,11 @@ if (NOT BLT_LOADED)
   mark_as_advanced(BLT_LOADED)
 
   set( BLT_ROOT_DIR ${CMAKE_CURRENT_LIST_DIR} CACHE PATH "" FORCE )
+  
+  # if an explicit build dir was not specified, set a default.
+  if( NOT BLT_BUILD_DIR )
```

Comment:
#84 writes this more simply as:
```cmake
 set( BLT_BUILD_DIR ${PROJECT_BINARY_DIR}/blt CACHE PATH "" ) 
```
Note: The ``if()`` check is missing, as is the ``FORCE`` keyword.
I checked the documentation to make sure this is correct.

from https://cmake.org/cmake/help/v3.8/command/set.html:
> Set the given cache <variable> (cache entry). Since cache entries are meant to provide user-settable values this does not overwrite existing cache entries by default. Use the FORCE option to overwrite existing entries.

Reply:
Yes #84 is simpler, but it doesn't allow you to actually pass in your own BLT_BUILD_DIR.

I thought we wanted to support a user settable build loc?

Reply:
Yes -- both versions should allow you to pass in a user-settable parameter.  @robinson96's version is shorter. 

Not sure if I was clear above.  According to the docs, 
```cmake
 set( BLT_BUILD_DIR ${PROJECT_BINARY_DIR}/blt CACHE PATH "" ) 
```
should do the following:
* if ``BLT_BUILD_DIR`` is not already a cache variable, it will add it to the cache and set its value to ``${PROJECT_BINARY_DIR}/blt``
* if it is already a cache variable, the value of ``BLT_BUILD_DIR`` will not be modified by this call (since the statement is missing the ``FORCE`` keyword)

It should have the same effect as your three liner.

Reply:
Here is a simple example to corfirm

```cmake
# -- cache.cmake
set(FOO "foo-cache-value" CACHE PATH "")
set(BAR "bar-cache-value" CACHE PATH "")
set(BAZ "baz-cache-value" CACHE PATH "")

#-- CMakeLists.txt
project(test-var)

foreach(v FOO BAR BAZ)
  message(STATUS "${v} before -- ${${v}}")
endforeach()

set(FOO "foo-in-cmake-lists" CACHE PATH "" )
set(BAR "bar-in-cmake-lists" CACHE PATH "" FORCE)
set(BAZ "baz-in-cmake-lists" )

foreach(v FOO BAR BAZ)
  message(STATUS "${v} after -- ${${v}}")
endforeach()
```
Running ``cmake -C ../cache.cmake`` yields the following
```bash
-- FOO after -- foo-cache-value
-- BAR after -- bar-in-cmake-lists    # overridden due to CACHE FORCE
-- BAZ after -- baz-in-cmake-lists    # overridden due to missing CACHE
```

Reply:
got it - I did not realize that was the case.  I guess you can tell I don't override things using that mechanism.  Do we do this anywhere else in BLT?

The behavior for the 3 line form seems much clearer to me - what do other folks think?

Also, here is some more info about what happens when things are already set:

>  It is possible for the cache entry to exist prior to the call but have no type set if it was created on the cmake(1) command line by a user through the -D<var>=<value> option without specifying a type. In this case the set command will add the type. Furthermore, if the <type> is PATH or FILEPATH and the <value> provided on the command line is a relative path, then the set command will treat the path as relative to the current working directory and convert it to an absolute path.

I have been always using as paths, but is that the logic we want for relative paths? 
Sounds confusing ...